docs/src/apimanual.md

```diff
@@ -470,58 +485,128 @@ non-global tree search solvers like
 ## A complete example: solving a knapsack problem
 
 [ needs formatting help, doc tests ]
-
+We first need to select a solver supporting the given problem (see
+[`supports`](@ref) and [`supports_constraint`](@ref)). In this example, we
+want to solve a binary-constrained knapsack problem:
+`max c'x: w'x <= C, x binary`. Suppose we choose GLPK:
 ```julia
-using MathOptInterface
-const MOI = MathOptInterface
 using GLPK
+optimizer = GLPK.Optimizer()
+```
+we can check that it supports the objective as follows:
+```jldoctest knapsack; setup = :(optimizer = MOI.Utilities.MockOptimizer(MOI.Utilities.Model{Float64}()); MOI.Utilities.set_mock_optimize!(optimizer, mock -> MOI.Utilities.mock_optimize!(mock, ones(3))))
+MOI.supports(optimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}())
 
-# Solves the binary-constrained knapsack problem:
-# max c'x: w'x <= C, x binary using GLPK.
+# output
 
+true
+```
+we can check that it supports the knapsack constraint as follows:
```

Comment:
I don't know if showing how to check if the solver supports the constraints is helpful at the start of the example. This isn't part of the normal workflow.

If you know that GLPK supports the problem type, then you just use it. If you don't know, then this approach of calling `supports_constraint` isn't especially helpful. Are you supposed to go through this exercise with each possible solver until you find one that works?

Reply:
Now, it was just to illustrate the use of `supports_constraint`. It might not be the best place for a first example

Reply:
Readers will be looking at this section as a first example of how to use MOI. It's good to have an illustration of `supports_constraint`, but it could be in a more advanced example.

Reply:
Yes, I kind of forgot why I included it. Removed :)